packages/lodestar/src/api/impl/beacon/state/state.ts

```diff
@@ -44,17 +45,45 @@ export class BeaconStateApi implements IBeaconStateApi {
     };
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async getStateValidators(stateId: StateId, filters?: IValidatorFilters): Promise<phase0.ValidatorResponse[]> {
     const state = await resolveStateId(this.chain, this.db, stateId);
     if (!state) {
       throw new StateNotFound();
     }
-    //TODO: implement filters
+
+    const currentEpoch = getCurrentEpoch(this.config, state);
+
+    const validators: phase0.ValidatorResponse[] = [];
+    if (filters?.indices) {
+      for (const id of filters.indices) {
+        const validatorIndex = getStateValidatorIndex(id, state, this.chain);
+        if (validatorIndex) {
+          const validatorResponse = toValidatorResponse(
+            validatorIndex,
+            state.validators[validatorIndex],
+            state.balances[validatorIndex],
+            currentEpoch
+          );
+          validators.push(validatorResponse);
+        }
+      }
+      return validators;
+    }
+    if (filters?.statuses) {
```

Comment:
what if both indices and status are passed?

Reply:
ah yeah i meant to implement that, but forgot.  i will take care of that